src/services.js

```diff
@@ -70,6 +70,15 @@ export function activityForDoc(nodeOrDoc) {
 }
 
 /**
+ * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
+ * @return {!../extensions/amp-form/0.1/amp-form.AmpFormService}
+ */
+export function ampFormServiceForDoc(nodeOrDoc) {
+  return /** @type {!../extensions/amp-form/0.1/amp-form.AmpFormService} */ (
```

Comment:
This is not so much about custom element registration as whether or not the service is hosted in the extension. `getElementServiceForDoc` (probably not the best name) handles the case when the service is deployed via extension, which I believe is the case here. Among other things, it denies the service promise if the extension has not been included, which makes sense since the service will never come.